docs/life-of-a-feature.md

```diff
@@ -26,3 +26,51 @@
 2. Testing and feedback.
    - Before a feature is released, get some peers to try out these new features by following public documentation. Receive feedback and iterate.
 3. Release and announcement. Cheers!
+
+
+## Branch Management
```

Comment:
Shall we have it as part of (probably top of) https://github.com/bytebase/bytebase/blob/main/docs/commit-guide.md ?

Life of a feature doc means to be high-level philosophy.

Reply:
Then we should change `commit-guide.md` to something like `dev-guide.md`?
Others LGTM.

Reply:
dev-guide.md SGTM, or github-dev-guide.md / git-dev-guide.md

Reply:
OK, I will go with dev-guide.md for now, which is intuitive enough currently.